erpnext/crm/doctype/lead/lead.js

```diff
@@ -28,6 +29,25 @@ erpnext.LeadController = frappe.ui.form.Controller.extend({
 		this.frm.set_query("contact_by", function (doc, cdt, cdn) {
 			return { query: "frappe.core.doctype.user.user.user_query" }
 		});
+
+		if (this.frm.doc.region) {
+			this.frm.set_query("territory", () => {
+				return {
+					query: "erpnext.crm.doctype.lead.lead.filter_territory",
+					filters: {
+						region: this.frm.doc.region
```

Comment:
If you're using arrow functions, you might want to retrieve the original `this` object:

```suggestion
						region: me.frm.doc.region
```